fix(gen): avoid faulty warnings for `number` and `integer` types

danilvpetrov opened the pull request:
This PR fixes faulty generator warnings when generating the code for `number` and `integer` types declared in the input YAML.

The example of input YAML: 
```yaml

  ... 
            ratio:
                description: The ratio of something.
                type: number
                minimum: 0
                maximum: 1
            total:
                description: The total of something.
                type: integer
                minimum: 0
   ...
```

results in the following warnings:
```
WARN	schemagen	Float validator cannot be applied to generated type and will be ignored	{"at": "file:///somefile.yml:xx:xx", "type": "number", "format": "", "go_type": "float64"}
WARN	schemagen	Int validator cannot be applied to generated type and will be ignored	{"at": "file:///somefile.yaml:xx:xx", "type": "integer", "format": "", "go_type": "int"}
```

tdakkota commented on gen/schema_gen.go:
```diff
@@ -314,7 +314,7 @@ func (g *schemaGen) generate2(name string, schema *jsonschema.Schema) (ret *ir.T
 			if t.Validators.Int.Set() {
 				switch t.Primitive {
-				case ir.String, ir.ByteSlice:
+				case ir.Int, ir.Int8, ir.Int16, ir.Int32, ir.Int64, ir.ByteSlice:
```
```suggestion
				case ir.Int,
					ir.Int8,
					ir.Int16,
					ir.Int32,
					ir.Int64,
					ir.Uint,
					ir.Uint8,
					ir.Uint16,
					ir.Uint32,
					ir.Uint64:
```

danilvpetrov replied:
Fixed in https://github.com/ogen-go/ogen/pull/735/commits/dc1c5ea0bda977e6001a117e5bfb7e3b72e09d2c.

tdakkota commented on gen/schema_gen.go:
```diff
@@ -325,7 +325,7 @@ func (g *schemaGen) generate2(name string, schema *jsonschema.Schema) (ret *ir.T
 			if t.Validators.Float.Set() {
 				switch t.Primitive {
-				case ir.String, ir.ByteSlice:
+				case ir.Float32, ir.Float64, ir.ByteSlice:
```
```suggestion
				case ir.Float32, ir.Float64:
```

danilvpetrov replied:
Fixed in https://github.com/ogen-go/ogen/pull/735/commits/dc1c5ea0bda977e6001a117e5bfb7e3b72e09d2cf.

tdakkota reviewed (changes requested):
Thank you for finding and fixing this!

Could you rewrite the commit to one that follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), please?

danilvpetrov commented:
Done, rewrote a commit message in https://github.com/ogen-go/ogen/pull/735/commits/dc1c5ea0bda977e6001a117e5bfb7e3b72e09d2c with all suggested changes incorporated. Also changed the title of the PR to be consistent with the commit message. Hopefully that works for you.
